sdks/java/core/src/main/java/org/apache/beam/sdk/coders/StructuredCoder.java

```diff
@@ -33,18 +32,23 @@
  * <p>To extend {@link StructuredCoder}, override the following methods as appropriate:
  *
  * <ul>
- *   <li>{@link #getComponents}: the default implementation returns {@link #getCoderArguments}.</li>
- *   <li>{@link #getEncodedElementByteSize} and
- *       {@link #isRegisterByteSizeObserverCheap}: the
+ *   <li>{@link #getComponents}: the default implementation returns {@link #getCoderArguments}.
+ *   <li>{@link #getEncodedElementByteSize} and {@link #isRegisterByteSizeObserverCheap}: the
  *       default implementation encodes values to bytes and counts the bytes, which is considered
- *       expensive.</li>
+ *       expensive.
  * </ul>
  */
 public abstract class StructuredCoder<T> extends Coder<T> {
+  /**
+   * Create a new {@link StructuredCoder}.
+   */
   protected StructuredCoder() {}
 
   /**
    * Returns the list of {@link Coder Coders} that are components of this {@link Coder}.
+   *
+   * <p>The default components will be equal to the value returned by {@link #getCoderArguments()},
+   * or the empty {@link List} if {@link #getCoderArguments()} returns {@code null}.
```

Comment:
I've updated the doc to not permit nulls. We'll still handle nulls for at least the time being (i.e. probably this MV), but users _should not_ return null from the method.